src/common/Button/Button.tsx

```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+
+interface buttonProps {
+	text: string;
+	onClick?(): void | React.MouseEvent<HTMLButtonElement>;
```

Comment:
This typing means that `onClick` doesn't accept any arguments and returns nothing or `React.MouseEvent<HTMLButtonElement>`, but this is not accurate because `onClick` accepts `React.MouseEvent<HTMLButtonElement>` and returns nothing, so I would suggest to change it to:

```suggestion
	onClick?: (e: MouseEvent<HTMLButtonElement>) => void;
```